config/main.py

```diff
@@ -1444,132 +1548,125 @@ def speed(ctx, interface_name, interface_speed, verbose):
+    # Get list of interfaces to be added
+    add_ports = parse_platform_json_file(BREAKOUT_CFG_FILE, interface_name, target_brkout_mode=target_brkout_mode)
+    add_intf_dict = {intf: add_ports[intf]["speed"] for intf in add_ports}
 
-            """
-            Added for additional Testing. will remove Later
-            """
-            config_db = ConfigDBConnector()
-            config_db.connect()
-            for intf in port_dict.keys():
-                click.secho("before modification for port {}".format(intf),  fg="cyan", underline=True)
-                print(config_db.get_entry('PORT',intf))
-                click.secho("after modification for port {}".format(intf),  fg="cyan", underline=True)
-                config_db.mod_entry('PORT',intf, port_dict[intf])
-                print(config_db.get_entry('PORT',intf))
+    if add_intf_dict:
+        click.echo("Ports to be added : \n {}".format(json.dumps(add_intf_dict, indent=4)))
+    else:
+        click.secho("[ERROR] port_dict is None!", fg='red')
+        raise click.Abort()
 
-            click.secho("PREV Breakout mode for port {}".format(interface_name),  fg="cyan", underline=True)
-            print(config_db.get_entry('BREAKOUT_CFG',interface_name))
+    """ Special Case: Dont delete those ports  where the current mode and speed of the parent port
+                      remains unchanged to limit the traffic impact """
 
-            brkout_cfg_keys = config_db.get_keys('BREAKOUT_CFG')
-            if interface_name.decode("utf-8") in  brkout_cfg_keys:
-                config_db.set_entry("BREAKOUT_CFG", interface_name,{'brkout_mode': target_brkout_mode})
+    click.secho("\nAfter running Logic to limit the impact", fg="cyan", underline=True)
+    matched_item = [intf for intf, speed in del_intf_dict.items() if intf in add_intf_dict.keys() and speed == add_intf_dict[intf]]
 
-            click.secho("CUR Breakout mode for port {}".format(interface_name),  fg="cyan", underline=True)
-            print(config_db.get_entry('BREAKOUT_CFG',interface_name))
+    # Remove the interface which remains unchanged from both del_intf_dict and add_intf_dict
+    map(del_intf_dict.pop, matched_item)
+    map(add_intf_dict.pop, matched_item)
 
+    click.secho("\nFinal list of ports to be deleted : \n {} \nFinal list of ports to be added :  \n {}".format(json.dumps(del_intf_dict, indent=4), json.dumps(add_intf_dict, indent=4), fg='green', blink=True))
+    if len(add_intf_dict.keys()) == 0:
+        click.secho("[ERROR] add_intf_dict is None! No interfaces are there to be added", fg='red')
+        raise click.Abort()
 
-    else:
-        click.secho("[ERROR] {} is not a Parent port. So, Breakout Mode is not available on this port".format(interface_name), fg='red')
+    port_dict = {}
+    for intf in add_intf_dict:
+        if intf in add_ports.keys():
+            port_dict[intf] = add_ports[intf]
+
+    # writing JSON object
+    with open('new_port_config.json', 'w') as f:
```

Comment:
Is this for Debug ??

Reply:
yeah, I remember. It was decided to give an option to user to save this file if the user wants to. However,  https://github.com/zhenggen-xu/sonic-buildimage/blob/sonic-cfg-mgmt/platform/vs/tests/breakout/test_breakout_cli.py#L32  is currently using this file for comparison test. So, if we provide an option like `--save` to this `cli` , we should add that option to the above mentioned file (https://github.com/zhenggen-xu/sonic-buildimage/blob/sonic-cfg-mgmt/platform/vs/tests/breakout/test_breakout_cli.py#L27) too as 'new_port_config.json' is needed to verify the test case.